src/search/search_algorithms/eager_search.cc

```diff
@@ -242,7 +243,7 @@ SearchStatus EagerSearch::step() {
             }
         } else if (succ_node.get_g() > node->get_g() + get_adjusted_cost(op)) {
             // We found a new cheapest path to an open or closed state.
-            if (reopen_closed_nodes) {
+            if (succ_node.is_open() || reopen_closed_nodes) {
```

Comment:
Would the logic be clearer if we changed from
```c++
if (is_open || reopen) {
    if (is_closed) {
        // Handle closed node that should be reopened
    } else {
        // Handle open node that should be updated
    }
    // Reinsert node into open list
} else {
    // Handle closed node that should not be reopened
}
```
to 
```c++
if (is_open) {
        // Handle open node that should be updated
        // Reinsert node into open list
} else if (is_closed && reopen) {
        // Handle closed node that should be reopened
        // Reinsert node into open list
} else {
    assert(is_closed && !reopen);
    // Handle closed node that should not be reopened
}
```
or 

```c++
if (is_open) {
        // Handle open node that should be updated
        // Reinsert node into open list
} else {
    assert(is_closed)
    if (reopen) {
        // Handle closed node that should be reopened
        // Reinsert node into open list
    } else {
        // Handle closed node that should not be reopened
    }
}
```
To avoid duplicating the code for reinsertion, we could extract this into a function.